compiler/src/iree/compiler/Codegen/Common/FastMathPatterns.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/FastMathPatterns.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/PatternMatch.h"
+
+namespace mlir::iree_compiler {
+
+namespace {
+// Pattern to lower math.erf to its device lib implementation
+// (from
+// https://github.com/ROCm/llvm-project/blob/amd-staging/amd/device-libs/ocml/src/erfF.cl#L11)
+struct FastErfPattern : public OpRewritePattern<math::ErfOp> {
+  using OpRewritePattern<math::ErfOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(math::ErfOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.getOperand();
+    Type resultType = op.getType();
+
+    // Erf only supports f32.
+    if (!resultType.isF32()) {
+      return rewriter.notifyMatchFailure(op, "Result only supports f32");
+    }
+
+    // Create constants
```

Comment:
style: proper punctuation on comments
```suggestion
    // Create constants.
```

(here and elsewhere in this file)

https://google.github.io/styleguide/cppguide.html#Punctuation_Spelling_and_Grammar